quartz/plugins/transformers/lastmod.ts

```diff
@@ -36,23 +58,24 @@ export const CreatedModifiedDate: QuartzTransformerPlugin<Partial<Options>> = (u
         () => {
           let repo: Repository | undefined = undefined
           return async (_tree, file) => {
-            let created: MaybeDate = undefined
-            let modified: MaybeDate = undefined
-            let published: MaybeDate = undefined
+            let created: DateTime | undefined = undefined
+            let modified: DateTime | undefined = undefined
+            let published: DateTime | undefined = undefined
 
             const fp = file.data.filePath!
             const fullFp = path.isAbsolute(fp) ? fp : path.posix.join(file.cwd, fp)
             for (const source of opts.priority) {
               if (source === "filesystem") {
                 const st = await fs.promises.stat(fullFp)
-                created ||= st.birthtimeMs
-                modified ||= st.mtimeMs
+                // birthtime can be 0 on some filesystems, so default to the earlier of ctime/mtime
+                created ||= DateTime.fromMillis(st.birthtimeMs || Math.min(st.ctimeMs, st.mtimeMs))
+                modified ||= DateTime.fromMillis(st.mtimeMs)
```

Comment:
this would probably add a bit of performance regression in terms of parsing time.

Can you run a quick bench comparing before and after for build time?

Reply:
will do. but FWIW, the net computation is for the most part unchanged for this codepath. it goes from `coerceDate(st.birthtimeMs)`, which is basically `new Date(st.birthtimeMs)`, to `DateTime.fromMillis(st.birthtimeMs...)`

Reply:
Benchmark results:

- using [hyperfine](https://github.com/sharkdp/hyperfine) w/ `hyperfine -w5 'npx quartz build'` 
- using https://github.com/RobertTalbert/discretecs-vault as the `content` directory (with `START HERE.md` renamed to `index.md`)
- using default `quartz.config.ts` except for `Plugin.CreatedModifiedDate({ priority: ["filesystem"] })`

For `v4` branch (commit ff9e60a)

```
Benchmark 1: npx quartz build
  Time (mean ± σ):      5.006 s ±  0.026 s    [User: 7.043 s, System: 1.184 s]
  Range (min … max):    4.953 s …  5.039 s    10 runs
```

For this PR (commit 5e7580f)

```
Benchmark 1: npx quartz build
  Time (mean ± σ):      5.142 s ±  0.069 s    [User: 7.132 s, System: 1.204 s]
  Range (min … max):    5.024 s …  5.269 s    10 runs
```

The difference in build time is quite small but *seems* to be consistent. I *assume* it's the overhead of Luxon vs native `Date`